
+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/bcmdrivers/broadcom/char/bcmlibs/impl1/bcm_pktfwd.c
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/bcmdrivers/broadcom/char/bcmlibs/impl1/bcm_pktfwd.c

@@ -675,6 +675,7 @@
     printk("\t\t %4s %10s %10s", "prio", "tot_pkts", "tx_drops");
     printk(" %10s %10s", "cfp_pkts", "cfp_fwds");
     printk(" %10s %10s", "chn_pkts", "chn_fwds");
+    printk(" %10s", "slow_pkts");
     printk(" %10s %10s\n", "schedule", "complete");
 
     for (prio = 0U; prio < D3FWD_PRIO_MAX; ++prio)
@@ -683,6 +684,7 @@
         printk("\t\t %4u %10u %10u", prio, stats->tot_pkts, stats->tx_drops);
         printk(" %10u %10u", stats->cfp_pkts, stats->cfp_fwds);
         printk(" %10u %10u", stats->chn_pkts, stats->chn_fwds);
+        printk(" %10u", stats->slow_pkts);
         printk(" %10u %10u\n", stats->schedule, stats->complete);
     }
 

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/bcmdrivers/broadcom/include/bcm963xx/bcm_pktfwd.h
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/bcmdrivers/broadcom/include/bcm963xx/bcm_pktfwd.h

@@ -1985,6 +1985,8 @@
     uint32_t   cfp_fwds;    /* xfer count - bypass via WLCFP */
     uint32_t   chn_pkts;    /* pkts count - WLCFP miss use native chain_node */
     uint32_t   chn_fwds;    /* xfer count - WLCFP miss use native chain_node */
+    uint32_t   slow_pkts;   /* xfer count - WLCFP miss, no PKTC, fwd one pkt at
+                             * a time */
     uint32_t   tx_drops;    /* dropped packets */
     uint32_t   schedule;    /* wlan thread: wlif net device dispatch requests */
     uint32_t   complete;    /* wlan thread: wlif net device scheduled */

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/kernel/linux-4.1/net/bridge/br_device.c
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/kernel/linux-4.1/net/bridge/br_device.c

@@ -26,11 +26,11 @@
 #include <linux/blog.h>
 #endif
 #if defined(CONFIG_BCM_KF_WL)
-#if defined(PKTC)
+#if defined(PKTC) || defined(PKTC_TBL)
 #include <osl.h>
 #include <wl_pktc.h>
 extern unsigned long (*wl_pktc_req_hook)(int req_id, unsigned long param0, unsigned long param1, unsigned long param2);
-#endif /* PKTC */
+#endif /* PKTC || PKTC_TBL */
 #include <linux/bcm_skb_defines.h>
 #endif
 
@@ -104,7 +104,7 @@
 		blog_link(BRIDGEFDB, blog_ptr(skb), (void*)dst, BLOG_PARAM1_DSTFDB, 0);
 		blog_unlock();
 #if defined(CONFIG_BCM_KF_WL)
-#if defined(PKTC)
+#if defined(PKTC) || defined(PKTC_TBL)
 		if (wl_pktc_req_hook && (dst->dst != NULL) &&
 			(BLOG_GET_PHYTYPE(dst->dst->dev->path.hw_port_type) == BLOG_WLANPHY) && 
 			wl_pktc_req_hook(PKTC_TBL_GET_TX_MODE, 0, 0, 0))
@@ -135,8 +135,8 @@
 				}
 			}
 		}
-#endif
-#endif
+#endif /* defined(PKTC) || defined(PKTC_TBL) */
+#endif /* defined(CONFIG_BCM_KF_WL) */
 		br_deliver(dst->dst, skb);
 	}        
 #else

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/kernel/linux-4.1/net/bridge/br_input.c
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/kernel/linux-4.1/net/bridge/br_input.c

@@ -25,14 +25,14 @@
 #include <linux/blog.h>
 #endif
 #if defined(CONFIG_BCM_KF_WL)
-#if defined(PKTC)
+#if defined(PKTC)  || defined(PKTC_TBL)
 #include <osl.h>
 #include <wl_pktc.h>
 unsigned long (*wl_pktc_req_hook)(int req_id, unsigned long param0, unsigned long param1, unsigned long param2) = NULL;
 EXPORT_SYMBOL(wl_pktc_req_hook);
 unsigned long (*dhd_pktc_req_hook)(int req_id, unsigned long param0, unsigned long param1, unsigned long param2) = NULL;
 EXPORT_SYMBOL(dhd_pktc_req_hook);
-#endif /* PKTC */
+#endif /* defined(PKTC)  || defined(PKTC_TBL) */
 #include <linux/bcm_skb_defines.h>
 #endif
 
@@ -269,7 +269,7 @@
 
 		blog_unlock();
 
-#if defined(PKTC)
+#if defined(PKTC)  || defined(PKTC_TBL)
 		/* wlan pktc */
 		if ((dst != NULL) && (dst->dst != NULL) && (!dst->is_local)) {
 #if defined(CONFIG_BCM_KF_WL)
@@ -394,7 +394,7 @@
 #endif
 		}
 next:
-#endif /* PKTC */
+#endif /* defined(PKTC)  || defined(PKTC_TBL) */
 		if ((dst != NULL) && dst->is_local) {
 			skb2 = skb;
 			/* Do not forward the packet since it's local. */
